### ISSUE TITLE
Compile OpenSSL in Visual Studio

### ISSUE THREAD

**Issue by MChelik** [openssl/openssl#1542]
First of all, maybe this is not openSSL problem but please help me, because I searched a lot but couldn't solve my damn problem. Thank you all.

I have
- openSSL 1.1.0
- ActivePerl 5.24 x86
- NASM x86 2.12.02
- Visual Studio 2010 Ultimate 10.0.30309
- Windows 7 Ultimate x64

I extracted source code, opened Visual Studio Command Prompt (2010) and entered following command:
c:\openSSL>**perl Configure VC-WIN32 --prefix:C:\openSSL-win32**

but I get this error:

> It looks like you don't have either nmake.exe or dmake.exe on your PATH, so you will not be able to execute the commands from a Makefile. You can install dmake.exe with the Perl Package Manager by running: ppm install dmake

I surely have nmake in VS Command Prompt. How can I tell Perl to see my nmake in VS directory?

P.S. These paths are my in my %PATH%:
- C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\bin
- C:\Perl\bin
- C:\Program Files (x86)\NASM

My full %PATH% is [here](http://pastebin.com/index/L5iKuFn5).

**Comment by MChelik** [comment]
Update: I uninstalled ActivePerl and install Strawberry Perl instead. Now it configures without error but **do_ms.bat** file is missing :(

**Comment by mattcaswell** [comment]
The build procedure for 1.1.0 has changed and you no longer need to use do_ms.bat. Please read INSTALL and NOTES.WIN. In brief:
```
    $ perl Configure VC-WIN32 --prefix:C:\openSSL-win32
    $ nmake
    $ nmake test
    $ nmake install
````

**Comment by MChelik** [comment]
Now when I run nmake I receive link error:

> "C:\Perl\perl\bin\perl.exe" "util\mkdef.pl" "crypto" 32 > libcrypto-1_1.def
> "C:\Perl\perl\bin\perl.exe" -i.tmp -pe "s|^LIBRARY\s+crypto32|LIBRARY libcrypto-1_1|;" libcrypto-1_1.def
> DEL libcrypto-1_1.def.tmp
> "C:\Perl\perl\bin\perl.exe" "util\mkrc.pl" libcrypto-1_1.dll > libcrypto-1_1.rc
> rc /folibcrypto-1_1.res libcrypto-1_1.rc
> Microsoft (R) Windows (R) Resource Compiler Version 6.1.7600.16385
> Copyright (C) Microsoft Corporation.  All rights reserved.
> 
> IF EXIST libcrypto-1_1.dll.manifest DEL /F /Q libcrypto-1_1.dll.manifest
> 
> link /nologo /debug /dll  /implib:libcrypto.lib /out:libcrypto-1_1.dll /
> def:libcrypto-1_1.def @C:\Users\MUSTAF~1.CEL\AppData\Local\Temp\nmB09.tmp || (DE
> L /Q libcrypto.* libcrypto-1_1.* && EXIT 1)
>    Creating library libcrypto.lib and object libcrypto.exp
> LINK : fatal error LNK1123: failure during conversion to COFF: file invalid or corrupt
> NMAKE : fatal error U1077: 'link' : return code '0x1'
> Stop.

**Comment by mattcaswell** [comment]
Hmmmmm....ping @levitte - any ideas?

**Comment by levitte** [comment]
No idea for the moment.  Since we don't know what files are lying around, I suggest a clean pristine unpacking of the source tarball and start over.

**Comment by MChelik** [comment]
I extracted fresh files and did it from scratch but result was the same (even with no-asm configuration). Installed Windows SDK, result was the same. Try x64 result was the same. The only thing that I didn't try was building in the North Pole (temperature effects linker). Do you have any idea guys? :(

**Comment by levitte** [comment]
I did a quick search, and found some interesting stuff:

https://msdn.microsoft.com/en-us/library/7dz62kfh.aspx

http://stackoverflow.com/questions/10888391/error-link-fatal-error-lnk1123-failure-during-conversion-to-coff-file-inval

They seem to suggest that updating your VS2010 installation to SP1 would be a good idea.

**Comment by levitte** [comment]
How did it go?

**Comment by MChelik** [comment]
I just opened to comment about my result, then I saw you comment. Thank you for asking. I installed VS 2010 SP1 and all errors gone. After _nmake_, I did _nmake test_ that passed and finally did _nmake install_ that copied some files to my _Program Files_. There is only two DLLs: _libcrypto-1_1.dll_ and _libssl-1_1.dll_. If that is correct, my problem is solved.
Thank you very much levitte. You are awesome and kind.
